nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
It would indeed be nice to avoid this kind of check by protecting us against having multiple instances, another way to do that would be to provide `RequiredFieldValidator` as a singleton. 

The behavior in our case would be equivalent, only the API will change (`new RequiredFieldValidator()` vs `RequiredFieldValidator.instance()`), but I find the intent clearer for the user when providing a singleton rather than overriding `equals` and `hashCode`.

Which one do you prefer?